hardik/cart/cart_with_object.rb

```diff
@@ -0,0 +1,149 @@
+# Shopping Cart Class
+class Cart
+  # Constructor set at default position
+  # ------------------------------------------
+  def initialize
+    @products = Array.new(0)
+    @cart_item = Array.new(0)
+  end
+  # ------------------------------------------
+
+  # add product for shopping
+  # ------------------------------------------
+  def add_product
+    print "Enter the name of product:\t"
+    name = gets.chomp
+    print "Enter the quantity of product:\t"
+    quantity = gets.chomp.to_i
+    print "Enter the price of product:\t"
+    price = gets.chomp.to_i
+    @products.push({ 'name' => name, 'quantity' => quantity, 'price' => price })
+  end
+  # ------------------------------------------
+
+  # update product
+  # ------------------------------------------
+  def update_product
+    puts 'Enter id of product'
+    id = gets.chomp.to_i
+    print "Enter name of product:\t"
+    name = gets.chomp
+    print "Enter quantity of product:\t"
+    quantity = gets.chomp
+    print "Enter price of product:\t"
+    price = gets.chomp
+    @products[id] = { 'name' => name, 'quantity' => quantity, 'price' => price }
+  end
+  # ------------------------------------------
+
+  # view cart
+  # ------------------------------------------
+  def view_product
+    puts "Id \t Name \t Quantity \t Price"
+    puts '------------------------------------------'
```

Comment:
You can store these dashes string in an object and use it here to avoid inconsistency and increase readability